osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -138,14 +217,71 @@ public void handleGetFirmwareVersionResponse(final List<FirmwareVersionDto> firm
+                    .withIds(ids)
+                    .withResult(result)
+                    .withOsgpException(osgpException)
+                    .withDataObject((Serializable) firmwareVersions)
+                    .withMessagePriority(messagePriority)
+                    .build();
+            this.webServiceResponseMessageSender.send(responseMessage);
+        }
+    }
+
+    private boolean checkSsldPendingFirmwareUpdate(final CorrelationIds ids,
+            final List<FirmwareVersion> firmwareVersions) {
+
+        final String deviceIdentification = ids.getDeviceIdentification();
+        final String organisationIdentification = ids.getOrganisationIdentification();
+        final String correlationUid = ids.getCorrelationUid();
+
+        final Ssld ssld = this.ssldRepository.findByDeviceIdentification(deviceIdentification);
+        final SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = ssld.getSsldPendingFirmwareUpdate();
+        if (ssldPendingFirmwareUpdate == null) {
+            return false;
+        }
+        if (!organisationIdentification.equals(ssldPendingFirmwareUpdate.getOrganisationIdentification())) {
+            return false;
+        }
+        if (!correlationUid.equals(ssldPendingFirmwareUpdate.getCorrelationUid())) {
+            return false;
+        }
+
+        LOGGER.info("Found SSLD pending firmware update record for device identification: {}, {}.",
+                deviceIdentification, ssldPendingFirmwareUpdate);
+
+        final FirmwareModuleType expectedFirmwareModuleType = ssldPendingFirmwareUpdate.getFirmwareModuleType();
+        final String expectedFirmwareVersion = ssldPendingFirmwareUpdate.getFirmwareVersion();
+        boolean foundExpectedFirmwareVersion = false;
+
+        for (final FirmwareVersion firmwareVersion : firmwareVersions) {
+            if (expectedFirmwareModuleType.equals(firmwareVersion.getFirmwareModuleType())
+                    && expectedFirmwareVersion.equals(firmwareVersion.getVersion())) {
+                foundExpectedFirmwareVersion = true;
+                break;
+            }
+        }
+
+        if (foundExpectedFirmwareVersion) {
+            LOGGER.info(
+                    "Expected firmware version from SSLD pending firmware update record matches firmware version as retrieved from device identification: {}, firmware version: {}, firmware module type: {}.",
+                    deviceIdentification, expectedFirmwareVersion, expectedFirmwareModuleType);
+        } else {
+            LOGGER.error(
+                    "Expected firmware version from SSLD pending firmware update record does not match firmware version as retrieved from device identification: {}, expected firmware version: {}, expected firmware module type: {}, actual firmware version and module type list:",
+                    deviceIdentification, expectedFirmwareVersion, expectedFirmwareModuleType);
```

Comment:
Consider adding the firmwareVersions List to the LOGGER.error above, instead of looping through the records.